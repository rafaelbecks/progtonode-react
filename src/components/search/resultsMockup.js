const resultsMockup =  [
  {
    id: 79827,
    title: 'John Frusciante',
    uri: '/artist/79827-John-Frusciante',
    type: 'artist',
    thumb: 'https://img.discogs.com/Mmuzj0--YFOH7CmFJUPX5Bn3UH0=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-79827-1512814987-9404.jpeg.jpg'
  },
  {
    id: 758717,
    title: 'Gail Frusciante And Friends',
    uri: '/artist/758717-Gail-Frusciante-And-Friends',
    type: 'artist',
    thumb: 'https://img.discogs.com/lFkRroItQazYPc_lhkGoGqL0biQ=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-758717-1558659509-2617.png.jpg'
  },
  {
    id: 801387,
    title: 'Gail Frusciante',
    uri: '/artist/801387-Gail-Frusciante',
    type: 'artist',
    thumb: '',
  },
  {
    id: 716683,
    title: 'Joseph Frusciante',
    uri: '/artist/716683-Joseph-Frusciante',
    type: 'artist',
    thumb: 'https://img.discogs.com/s0ajcpWDndjhnRDGJqt-otu6MWk=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-716683-1376487230-5662.png.jpg'
  }
];

export default resultsMockup;
