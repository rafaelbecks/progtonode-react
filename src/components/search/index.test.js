import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from './index';
import resultsMockup from './resultsMockup';

describe('Search component test with Enzyme', () => {
  it('renders without crashing without props', () => {
    shallow(<Search />);
  });

  it('renders without crashing with search data array', () => {
    shallow(<Search searchResult={resultsMockup} />);
  });

  it('render list correctly', () => {
    const searchComponent = mount(<Search searchResult={resultsMockup} />);

    expect(searchComponent.find('li')).toHaveLength(4);
    expect(
      searchComponent
        .find('li')
        .first()
        .text(),
    ).toEqual(resultsMockup[0].title);
  });
});


// search, searchResult, getGraphData
