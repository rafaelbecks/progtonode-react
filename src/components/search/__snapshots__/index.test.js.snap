// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component test with Enzyme renders and match snapshot 1`] = `
<div
  className="sc-bdVaJa jFNXxd"
>
  <label>
    Busca un artista
  </label>
  <div
    className="sc-bwzfXH byFnul"
  >
    <input
      name="query"
      type="text"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Buscar
    </button>
  </div>
  <p>
    Resultados
  </p>
  <ul>
    <li
      onClick={[Function]}
    >
      <img
        alt="John Frusciante"
        src="https://img.discogs.com/Mmuzj0--YFOH7CmFJUPX5Bn3UH0=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-79827-1512814987-9404.jpeg.jpg"
      />
      John Frusciante
    </li>
    <li
      onClick={[Function]}
    >
      <img
        alt="Gail Frusciante And Friends"
        src="https://img.discogs.com/lFkRroItQazYPc_lhkGoGqL0biQ=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-758717-1558659509-2617.png.jpg"
      />
      Gail Frusciante And Friends
    </li>
    <li
      onClick={[Function]}
    >
      <img
        alt="Gail Frusciante"
        src=""
      />
      Gail Frusciante
    </li>
    <li
      onClick={[Function]}
    >
      <img
        alt="Joseph Frusciante"
        src="https://img.discogs.com/s0ajcpWDndjhnRDGJqt-otu6MWk=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-716683-1376487230-5662.png.jpg"
      />
      Joseph Frusciante
    </li>
  </ul>
</div>
`;
